spock-core/src/main/java/org/spockframework/mock/runtime/InteractionScope.java

```diff
@@ -41,9 +45,13 @@ public void addInteraction(final IMockInteraction interaction) {
       public Object accept(IMockInvocation invocation) {
         WrongInvocationOrderError wrongInvocationOrderError = null;
         if (currentExecutionZone > myRegistrationZone) {
-          wrongInvocationOrderError = new WrongInvocationOrderError(decorated, invocation);
+          wrongInvocationOrderError = new WrongInvocationOrderError(decorated, invocation, previousInvocations);
         }
         currentExecutionZone = myRegistrationZone;
+        if (previousInvocations.size() == MAX_PREVIOUS_INVOCATIONS) {
+          previousInvocations.remove();
```

Comment:
If you use `addFirst` and `removeLast` instead then you get the reverse order out of the box, if we change the name to `previousInvocationsInReverseOrder` then there shouldn't be cause for confusion.

Reply:
as you wish